### PR TITLE
frontera virtual env script

### PR DESCRIPTION
**Pull request by leilaroshan**
Added a script to create a virtual environment on Frontera CPU and GPU nodes.

### CODE REVIEW

File: build_venv_frontera.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+module reset 
+
+# Load modules
+module load python3/3.9
+module load cuda/12
+
+# create env
+# ---------
+source module.sh
```
Comment (kks32) [inline build_venv_frontera.sh:11]: since you already loaded the modules, why do oyou need this?

File: build_venv_frontera.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+python3 -m virtualenv venv
+source venv/bin/activate
+
+which python
```
Comment (kks32) [inline build_venv_frontera.sh:16]: remove this

File: build_venv_frontera.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+python3 test/test_pytorch.py 
+
+echo 'test_pytorch_cuda_gpu.py -> True if GPU'
+python3 test/test_pytorch_cuda_gpu.py
+
+echo 'test_torch_geometric.py -> no return if import successful'
+python3 test/test_torch_geometric.py
+
+echo 'waterdrop sample/Dataset -> no return if import successful'
+python3 -m gns.train --data_path="./gns-sample/WaterDropSample/dataset/" --model_path="../models/" --output_path="../output/"  -ntraining_steps=100
```
Comment (kks32) [inline build_venv_frontera.sh:40]: don't include this in the build env.

File: build_venv_frontera.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+pip install torch==2.0.1 torchvision==0.15.2 torchaudio==2.0.2 --index-url https://download.pytorch.org/whl/cu118
+pip install torch_geometric==2.4.0
+pip install pyg_lib torch_scatter torch_sparse torch_cluster 
+torch_spline_conv -f https://data.pyg.org/whl/torch-2.0.0+cu118.html
+pip install -r requirements.txt
+
+# test env
+# --------
+
+echo 'which python -> venv'
```
Comment (kks32) [inline build_venv_frontera.sh:27]: Add a command line argument and run tests only if it is set to true.

**Review by kks32 — CHANGES_REQUESTED** [review]

**Comment by leilaroshan** [comment]
The modifications have been applied to the "build_venv_frontera.sh" file.

File: build_venv_frontera.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+fi
+
+echo 'test_pytorch.py -> random tensor'
+python3 test/test_pytorch.py 
+
+echo 'test_pytorch_cuda_gpu.py -> True if GPU'
+python3 test/test_pytorch_cuda_gpu.py
+
+echo 'test_torch_geometric.py -> no return if import successful'
+python3 test/test_torch_geometric.py
```
[the remark this answers is not in the document]
Comment (leilaroshan) [reply build_venv_frontera.sh]: Now It's all moved to the "Testing" section.

File: build_venv_frontera.sh [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+pip install torch_geometric==2.4.0
+pip install pyg_lib torch_scatter torch_sparse torch_cluster 
+torch_spline_conv -f https://data.pyg.org/whl/torch-2.0.0+cu118.html
+pip install -r requirements.txt
+
+# test env
+# --------
+
+echo 'which python -> venv'
+which python
```
Comment (kks32) [inline build_venv_frontera.sh:25]:
```suggestion

```

File: build_venv_frontera.sh [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  
+else
+  echo "Skipping tests. To run tests, use the argument --run-tests=true"
+fi
+
+
+# Clean up
+# --------
+#rm -r venv
+
```
Comment (kks32) [inline build_venv_frontera.sh:51]:
```suggestion
```

**Review by kks32 — CHANGES_REQUESTED** [review]
Minor visual edits

**Review by kks32 — APPROVED** [review]
Great job @leilaroshan ! Congrats on your first PR!

**Pull request merged**